spark/src/main/java/org/apache/iceberg/spark/SparkTableUtil.java

```diff
@@ -621,6 +640,32 @@ public static void importSparkPartitions(
     }
   }
 
+  public static List<SparkPartition> filterPartitions(List<SparkPartition> partitions,
+                                                      Map<String, String> partitionFilter) {
+    if (partitionFilter.isEmpty()) {
+      // No partition filter arg
+
+      if (partitions.isEmpty()) {
+        throw new IllegalArgumentException("Cannot add files, no files found in the table.");
+      }
+      return partitions;
+    } else {
+      // Partition filter arg passed
+
+      List<SparkTableUtil.SparkPartition> filteredPartitions = partitions
+          .stream().filter(p -> p.getValues().entrySet().containsAll(partitionFilter.entrySet()))
```

Comment:
Do we do any escaping/normalization for column names and values?

Reply:
I'm not sure, but if we don't match here we don't match against the iceberg column names which means we'll have some issues later. I'll add a test to check

Reply:
Let me think about this too.